### PR TITLE
configmap manage

### PR DESCRIPTION
**Pull request by warjiang**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
add REST api for management of configmap

- /configmap: list all configmaps across namespaces
- /configmap/:namespace: list configmaps under the scope of specific namespace
- /configmap/:namespace/:name: return the detail of configmap

for edit&delete configmap, we'll reuse the unstructured api

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
- REST api for management of configmap
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
OK. I will put this on my agenda.

File: pkg/resource/configmap/list_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2017 The Kubernetes Authors.
```
Comment (RainbowMango) [inline pkg/resource/configmap/list_test.go:1]: Is this file copied from Kubernetes?

Comment (warjiang) [reply pkg/resource/configmap/list_test.go:1]: yes, should I follow the rules of lift: https://karmada.io/docs/contributor/lifted

Comment (RainbowMango) [reply pkg/resource/configmap/list_test.go:1]: Not really. It looks unnormal to just copy test files. 
Don't have to be that complex.
But can you add a comment right after the license to explain the source link?

Comment (warjiang) [reply pkg/resource/configmap/list_test.go]: Got it. After reviewing that test file again, it's no need to lift that file from `kubernetes/dashboard` repo, so I decided deleting that test file.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**